plugins/c8y_log_plugin/src/logfile_request.rs

```diff
@@ -0,0 +1,165 @@
+        SmartRestSetOperationToFailed, SmartRestSetOperationToSuccessful,
+    },
+};
+use mqtt_channel::{Connection, SinkExt};
+
+pub struct LogfileRequest {}
+
+impl TryIntoOperationStatusMessage for LogfileRequest {
+    /// returns a c8y message specifying to set log status to executing.
+    ///
+    /// example message: '501,c8y_LogfileRequest'
+    fn status_executing() -> Result<
+        c8y_smartrest::smartrest_serializer::SmartRest,
+        c8y_smartrest::error::SmartRestSerializerError,
+    > {
+        SmartRestSetOperationToExecuting::new(CumulocitySupportedOperations::C8yLogFileRequest)
+            .to_smartrest()
+    }
+
+    fn status_successful(
+        parameter: Option<String>,
+    ) -> Result<
+        c8y_smartrest::smartrest_serializer::SmartRest,
+        c8y_smartrest::error::SmartRestSerializerError,
+    > {
+        SmartRestSetOperationToSuccessful::new(CumulocitySupportedOperations::C8yLogFileRequest)
+            .with_response_parameter(&parameter.unwrap())
+            .to_smartrest()
+    }
+
+    fn status_failed(
+        failure_reason: String,
+    ) -> Result<
+        c8y_smartrest::smartrest_serializer::SmartRest,
+        c8y_smartrest::error::SmartRestSerializerError,
+    > {
+        SmartRestSetOperationToFailed::new(
+            CumulocitySupportedOperations::C8yLogFileRequest,
+            failure_reason,
+        )
+        .to_smartrest()
+    }
+}
+/// Reads tedge logs according to `SmartRestLogRequest`.
+///
+/// If needed, logs are concatenated.
+///
+/// Logs are sorted alphanumerically from oldest to newest.
+///
+/// # Examples
+///
+/// ```
+/// let smartrest_obj = SmartRestLogRequest::from_smartrest(
+///     "522,DeviceSerial,syslog,2021-01-01T00:00:00+0200,2021-01-10T00:00:00+0200,,1000",
+/// )
+/// .unwrap();
+///
+/// let log = read_tedge_system_logs(&smartrest_obj, "/var/log/tedge").unwrap();
+/// ```
+pub fn read_tedge_logs(
+    smartrest_obj: &SmartRestLogRequest,
+    plugin_config_path: &PathBuf,
+) -> Result<String, anyhow::Error> {
+    let plugin_config = PluginConfig::new(&plugin_config_path);
+    let mut output = String::new();
+
+    let mut files_to_send = Vec::new();
+    for files in &plugin_config.files {
+        let maybe_file_path = files.path.as_str(); // because it can be a glob pattern
+        let file_type = files.config_type.as_str();
+
+        if !file_type.eq(&smartrest_obj.log_type) {
+            continue;
+        }
+
+        // NOTE: According to the glob documentation paths are yielded in alphabetical order hence re-ordering is no longer required see:
+        // https://github.com/thin-edge/thin-edge.io/blob/0320741b109f50d1b0f7cda44e33dc31ba04902d/plugins/log_request_plugin/src/smartrest.rs#L24
+        dbg!(&maybe_file_path);
+        for entry in glob(maybe_file_path)? {
+            dbg!(&entry);
+            let file_path = entry?;
+            if let Some(dt_from_file) = get_datetime_from_file_path(&file_path) {
```

Comment:
I think this logic is broken as per the latest specification. The datetime must be the last update time and not expected to be in the name of the file itself.

Reply:
Deferred for later as agreed.